IBM/carbon-addons-iot-react
# 1282 table manage views dropdown

Closes #1282

**Summary**
As part of #1210 we need to create a custom dropdown component that can manage the table views. Once the different components of #1210 are finished a full table story example will be created.

**Change List (commits, features, bugs, etc)**

- Instead of adding additional complexity to the TableComponent itself a ViewDropdown component was implemented as an external table sub component that the application must pass in via a render prop (customToolbarContent).

- Created a new folder called TableSubcomponents for this purpose. The ViewDropdown component got its own folder with code, styles, tests. 

- Unlike the underlying Dropdown the ViewDropdown component is stateless. The rationale is that the Application will most likely want to be able to change the selected item dynamically. E.g. when a user saves a new view, that view should probably be selected automatically. 

- A new story for TableSubcomponents was created showing how to use the ViewDropdown in a an application that handles the state. 

- The ViewDropdown was implemented to be highly configurable by using a lighter version of the emerging "Overrides Pattern" used by Uber. https://medium.com/@dschnr/better-reusable-react-components-with-the-overrides-pattern-9eca2339f646

- A ViewDropdownItem component was created to handle the special rendering requirements. We are forced to use the itemToString function to show the selected item they way it was designed, that is the cause of some of the complexity of the ViewDropdownItem. See closed PR https://github.com/carbon-design-system/carbon/pull/5578

- Added unit tests

**Acceptance Test (how to verify the PR)**
1. Go to story https://deploy-preview-1296--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-tablesubcomponents--viewdropdown
2. Check that all items are rendered correctly
3. Check that there is a "View all" default
4. Check that "Save as new view" and "Manage views" are available
5. Select "Save as new view" and make sure the corresponding action is called but that the selected item is not changed.
6. Select "Manage views" and make sure the corresponding action is called but that the selected item is not changed.
7. Verify that the selected view has postfix "- edited"
8. Uncheck the knob activeViewEdited and verify that the postfix "- edited" is removed.
9. Select the option "My saved 3 with a very long name.." and make sure it is rendered correctly
10. Use the knob to change the wrapper width to '100px' and verify that the icons of selected item and the Manage Views is removed. Verify that the "View" prefix is removed for the selected item.
